monitors/test/index.js

```diff
@@ -106,4 +113,67 @@ contract("index.js", function(accounts) {
       assert.notEqual(spyLogLevel(spy, i), "error");
     }
   });
+
+  it("Correctly re-tries after failed execution loop", async function() {
```

Comment:
this test was a massive pain to get right. Our bots are so robust they never throw errors in the while loop. needed to be pretty creative to get _past_ the initial checks but throw in the while loop!

Reply:
Interesting way to test this, so this fails on the first iteration  when it tries to update EMP state right?

Reply:
yup. you can create the EMP client but the bot errors out when the emp state updates as the functions the client calls are not defined in this "invalid" emp.